app/src/main/java/org/mozilla/tv/firefox/settings/SettingsFragment.kt

```diff
@@ -123,7 +139,12 @@ class SettingsFragment : Fragment() {
     companion object {
         const val FRAGMENT_TAG = "settings"
 
-        @JvmStatic
-        fun create() = SettingsFragment()
+        fun newInstance(type: SettingsType): SettingsFragment {
```

Comment:
> This would be a good place to log an inconsistent state assertion to Sentry.

Same as above.  Although I wonder if we could enforce this with the type system, instead.  It seems like our `SettingsType` has two separate purposes: display on the overlay, and show some settings screen.  This approach is similar to one I took with Pocket, which in retrospect I think was a mistake.

What if we had an interface `SettingsScreen` or similar, that only some `SettingsType`s implemented?  Then we could enforce that a `SettingsScreen` instance is passed here?